packages/react-components/src/components/TroubleShootingGuideErrorBar.tsx

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { useEffect, useState } from 'react';
+import { Link, MessageBar, PrimaryButton, Stack, useTheme } from '@fluentui/react';
+import { useLocale } from '../localization';
+import { ErrorBarProps } from './ErrorBar';
+import { confirmButtonStyle, linkStyle, messageBarStyle } from './styles/TroubleShootingGuideErrorBar.styles';
+import {
+  DismissedError,
+  dismissError,
+  dropDismissalsForInactiveErrors,
+  errorsToShow,
+  messageBarIconProps,
+  messageBarType
+} from './utils';
+
+/**
+ * Strings for {@link _TroubleShootingGuideErrorBar}.
+ *
+ * @internal
+ */
+export interface _TroubleShootingGuideErrorBarStrings {
+  linkText?: string;
+  buttonText?: string;
+}
+
+/**
+ * Props for {@link _TroubleShootingGuideErrorBar}.
+ *
+ * @internal
+ */
+export interface _TroubleShootingGuideErrorBarProps extends ErrorBarProps {
+  /**
+   * permissions state for camera/microphone
+   */
+  permissionsState: {
+    camera: PermissionState;
+    microphone: PermissionState;
+  };
+  /**
+   * Callback you may provide to supply users with further steps to troubleshoot why they have been
+   * unable to grant your site the required permissions for the call.
+   *
+   * @example
+   * ```ts
+   * onPermissionsTroubleshootingClick: () =>
+   *  window.open('https://contoso.com/permissions-troubleshooting', '_blank');
+   * ```
+   *
+   * @remarks
+   * if this is not supplied, the composite will not show a 'further troubleshooting' link.
+   */
+  onPermissionsTroubleshootingClick?: (permissionsState: {
+    camera: PermissionState;
+    microphone: PermissionState;
+  }) => void;
+  /**
+   * Optional callback to supply users with further troubleshooting steps for network issues
+   * experienced when connecting to a call.
+   *
+   * @example
+   * ```ts
+   * onNetworkingTroubleShootingClick?: () =>
+   *  window.open('https://contoso.com/network-troubleshooting', '_blank');
+   * ```
+   *
+   * @remarks
+   * if this is not supplied, the composite will not show a 'network troubleshooting' link.
+   */
+  onNetworkingTroubleShootingClick?: () => void;
+  /**
+   * strings related to trouble shooting guidance link and dismiss button
+   */
+  troubleShootingGuideStrings: _TroubleShootingGuideErrorBarStrings;
+}
+
+/**
+ * @internal
+ * A component to show device Permission/network connection related errors, contains link that leads to further trouble shooting guide
+ */
+export const _TroubleShootingGuideErrorBar = (props: _TroubleShootingGuideErrorBarProps): JSX.Element => {
```

Comment:
troubleshooting is one word, so _TroubleShootingGuideErrorBar-> _TroubleshootingGuideErrorBar (and filename, props names etc.)